core/src/Kokkos_BitManipulation.hpp

```diff
@@ -283,22 +284,20 @@ struct PopCount<constant_evaluated, /*device=*/false> {
 #undef KOKKOS_IMPL_USE_GCC_BUILT_IN_FUNCTIONS
 
 template <class T>
-inline constexpr bool is_standard_unsigned_integer_type_v =
-    std::is_same_v<T, unsigned char> || std::is_same_v<T, unsigned short> ||
-    std::is_same_v<T, unsigned int> || std::is_same_v<T, unsigned long> ||
-    std::is_same_v<T, unsigned long long>;
+concept UnsignedInteger =
+    std::same_as<T, unsigned char> || std::same_as<T, unsigned short> ||
+    std::same_as<T, unsigned int> || std::same_as<T, unsigned long> ||
+    std::same_as<T, unsigned long long>;
```

Comment:
Can this be replaced with `std::unsigned_integral`?

Reply:
Sorry I collapsed https://github.com/kokkos/kokkos/pull/8509#discussion_r2394142170 when I marked it as resolved.
No the standard (link in the description) clearly states these are constrained to take [standard unsigned integer](https://eel.is/c++draft/basic.fundamental#def:type,standard_unsigned_integer)
You will see similar code if you browse the standard library headers on your system.

Reply:
Maybe,
```suggestion
concept StandardUnsignedInteger =
    std::same_as<T, unsigned char> || std::same_as<T, unsigned short> ||
    std::same_as<T, unsigned int> || std::same_as<T, unsigned long> ||
    std::same_as<T, unsigned long long>;
```
is a better name then.

Reply:
Applied.